app/src/main/java/usb/AndroidUSBInputStream.java

```diff
@@ -1,83 +1,72 @@
- * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF 
- * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
- */
 package usb;
 
-import java.io.IOException;
-import java.io.InputStream;
-
 import android.hardware.usb.UsbDeviceConnection;
 import android.hardware.usb.UsbEndpoint;
-import android.util.Log;
+
+import com.fpvout.digiview.dvr.DVR;
 
 /**
  * This class acts as a wrapper to read data from the USB Interface in Android
  * behaving like an {@code InputputStream} class.
  */
+import java.io.IOException;
+import java.io.InputStream;
+
 public class AndroidUSBInputStream extends InputStream {
 
-	private final String TAG = "USBInputStream";
-	// Constants.
-	private static final int OFFSET = 0;
 	private static final int READ_TIMEOUT = 100;
 
-	// Variables.
-	private UsbDeviceConnection usbConnection;
-
-	private UsbEndpoint receiveEndPoint;
+	private final UsbDeviceConnection usbConnection;
+	private final UsbEndpoint receiveEndPoint;
 	private final UsbEndpoint sendEndPoint;
 
 	private boolean working = false;
 
 
+	private DVR dvr;
+
 	/**
 	 * Class constructor. Instantiates a new {@code AndroidUSBInputStream}
 	 * object with the given parameters.
 	 *
 	 * @param readEndpoint The USB end point to use to read data from.
-	 * @param connection The USB connection to use to read data from.
-	 *
+	 * @param sendEndpoint The USB end point to use to sent data to.
+	 * @param connection   The USB connection to use to read data from.
 	 * @see UsbDeviceConnection
 	 * @see UsbEndpoint
 	 */
-	public AndroidUSBInputStream( UsbEndpoint readEndpoint, UsbEndpoint sendEndpoint, UsbDeviceConnection connection) {
+	public AndroidUSBInputStream( UsbEndpoint readEndpoint, UsbEndpoint sendEndpoint, UsbDeviceConnection connection, DVR dvr) {
 		this.usbConnection = connection;
 		this.receiveEndPoint = readEndpoint;
+		this.dvr = dvr;
 		this.sendEndPoint = sendEndpoint;
 	}
 
 	@Override
-	public int read() throws IOException {
+	public int read() {
 		byte[] buffer = new byte[131072];
-		return read(buffer, 0,  buffer.length);
+		return read(buffer, 0, buffer.length);
 	}
 
 	@Override
-	public int read(byte[] buffer, int offset, int length) throws IOException {
+	public int read(byte[] buffer, int offset, int length) {
 		int receivedBytes = usbConnection.bulkTransfer(receiveEndPoint, buffer, buffer.length, READ_TIMEOUT);
 		if (receivedBytes <= 0) {
 			// send magic packet again; Would be great to handle this in UsbMaskConnection directly...
-			Log.d(TAG, "received buffer empty, sending magic packet again...");
+			//Log.d(TAG, "received buffer empty, sending magic packet again...");
 			usbConnection.bulkTransfer(sendEndPoint, "RMVT".getBytes(), "RMVT".getBytes().length, 2000);
 			receivedBytes = usbConnection.bulkTransfer(receiveEndPoint, buffer, buffer.length, READ_TIMEOUT);
+		} else {
+			this.dvr.recordVideoDVR(buffer,offset , buffer.length);
```

Comment:
maybe passing a copy of the buffer would be better (to really split streams)

Reply:
will try that but i guess it will consume more memory.